formatter/src/main/java/org/togetherjava/formatter/CodeSectionFormatter.java

```diff
@@ -0,0 +1,396 @@
+package org.togetherjava.formatter;
+
+import org.togetherjava.formatter.tokenizer.Token;
+import org.togetherjava.formatter.tokenizer.TokenType;
+import org.togetherjava.formatter.util.SkippableLookaheadArrayDeque;
+import org.togetherjava.formatter.util.SkippableLookaheadQueue;
+
+import java.util.*;
+import java.util.function.Predicate;
+
+/**
+ * Formatter which specifically formats code tokens (that are part of a section)
+ *
+ * @author illuminator3
+ */
+class CodeSectionFormatter {
+    private final StringBuilder result = new StringBuilder();
+    private final SkippableLookaheadQueue<Token> queue;
+
+    private int indentation;
+    private boolean applyIndentation;
+    private int forLevel;
+    private int genericDepth;
+    private int enhancedLevel = -1;
+
+    CodeSectionFormatter(List<Token> tokens) {
+        this(new SkippableLookaheadArrayDeque<>(tokens));
+    }
+
+    CodeSectionFormatter(SkippableLookaheadQueue<Token> queue) {
+        this.queue = queue;
+
+        purgeWhitespaces(this.queue);
+    }
+
+    /**
+     * Removes all whitespaces from the given queue
+     *
+     * @param queue the queue to remove whitespaces from
+     * @author illuminator3
+     */
+    private static void purgeWhitespaces(Queue<Token> queue) {
+        queue.removeIf(t -> t.type() == TokenType.WHITESPACE);
+    }
+
+    /**
+     * Starts the formatting process
+     *
+     * @author illuminator3
+     */
+    void format() {
+        Token next;
+
+        while (!queue.isEmpty() && (next = queue.remove()) != null) {
+            updateIndentation(next.type());
+            applyIndentation();
+            consume(next);
+        }
+    }
+
+    /**
+     * Consumes the next token
+     *
+     * @param token token to consume
+     * @author illuminator3
+     */
+    private void consume(Token token) {
+        TokenType type = token.type();
+
+        if (checkGeneric(type)) {
+            result.append(token.content());
+
+            return;
+        }
+
+        if (genericDepth != 0) {
+            handleGeneric(token);
+
+            return;
+        }
```

Comment:
I dislike your usage of whitespace here, I think my eyes prefer this:
```suggestion
        if (checkGeneric(type)) {
            result.append(token.content());
            return;
        }

        if (genericDepth != 0) {
            handleGeneric(token);
            return;
        }
```

Reply:
subjective.